### DISCUSSION TITLE
How to download a binary from enterprise github release?

### DISCUSSION THREAD

**Discussion by chuan137** [zdharma-continuum/zinit#208] [Q&A]
from README, the `from` ice also support random domain, such as enterprise github site,
however I want to download a binary from releases, how can I do that? below example does not
work, because the command clones the repo.

>      zinit light-mode for \
>         from'self.hosted.github.domain' as'program' \
>         mv'exec* -> exec' pick'exec' \
>         myteam/myrepo

**Comment by vladdoster** [comment]
@chuan137 

Are you putting in the full FQDN?

**Comment by vladdoster** [comment]
@chuan137 Is this still an issue, or did you figure it out?

**Comment by chuan137** [comment]
hi @vladdoster, still an issue

I used the full FQDN

```bash
zinit wait lucid light-mode for \
    from'my.corp.github.domain/cc/k8s/releases' as'program' \
        mv'k8s* -> k8s' pick'k8s' \
        cc/k8s \
```

**Comment by vladdoster** [comment]
This is documented on the wiki, see section [`Private-Repositories`](https://zdharma-continuum.github.io/zinit/wiki/Private-Repositories/)

**Reply by vladdoster** [reply]
> This is documented on the wiki; see section [`Private-Repositories`](https://zdharma-continuum.github.io/zinit/wiki/Private-Repositories/)

@chuan137 Did the documentation I linked work? If so, please mark it as the answer.

**Reply by chuan137** [reply]
I don't see how to download a binary from git releases in the doc

**Comment by decayofmind** [comment]
Same problem here. If one is using GitHub enterprise and want to download binaries from releases, then it's not possible.

**Comment by decayofmind** [comment]
This potentially can be possible using [zinit-annex-readurl](https://github.com/zdharma-continuum/zinit-annex-readurl) with netrc, however if both curl (will need a flag to look into netrc) and wget (will look into netrc file by default) are installed, then the first one is used :/

**Comment by psprint** [comment]
you can put the domain into from'' ice, like so: from'
enterprise.gituhub.com' (I'm not sure if it's the correct domain).

On Tue, 18 Jul 2023 at 18:32, Roman ***@***.***> wrote:

> This potentially can be possible using zinit-annex-readurl
> <https://github.com/zdharma-continuum/zinit-annex-readurl> with netrc,
> however if both curl (will need a flag to look into netrc) and wget (will
> look into netrc file by default) are installed, then the first one is used
> :/
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/zdharma-continuum/zinit/discussions/208#discussioncomment-6483156>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ABOE4CBUCW2KRCJGGBFMXTTXQ4MI3ANCNFSM5SB4HZAA>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***
> com>
>


-- 
Best regards,
Sebastian Gniazdowski

**Reply by decayofmind** [reply]
@psprint I'm not sure if it's possible. As in the original request, the problem is where:

* Need to download a program from GitHub Releases
* GitHub instance is on-premise (GitHub enterprise) 
* Access requires a token auth (private repository case)

**Comment by psprint** [comment]
Ah, yes, there's conflict between "`from'gh-r'` vs. `from"<custom-domain"`. Could someone look at it? It's just a matter of leading the code towards accepting the custom domain (code that does this is already there) and adding a new ice like `gh-binary` that would trigger github-releases download (or via accepting a prefix to `from''`, like: `from'gh-r:enterprise.github.com'`).

**Comment by psprint** [comment]
I would take a look at this myself however I'm tired of my PRs waiting for a merge nearly from one year… 
:disappointed:
